scaladoc/src/dotty/tools/scaladoc/site/SidebarParser.scala

```diff
@@ -30,16 +31,31 @@ object Sidebar:
 
   private object RawInputTypeRef extends TypeReference[RawInput]
 
-  private def toSidebar(r: RawInput)(using CompilerContext): Sidebar = r match
+  private def toSidebar(r: RawInput, content: String | java.io.File)(using CompilerContext): Sidebar = r match
     case RawInput(title, page, index, subsection, dir, hidden) if page.nonEmpty && index.isEmpty && subsection.isEmpty() =>
+      val sidebarPath = content match
+        case s: String => Paths.get(s)
+        case f: java.io.File => f.toPath()
+      val basePath = sidebarPath.getParent().resolve("_docs")
+      val pagePath = basePath.resolve(page)
+      if !Files.exists(pagePath) then
+        report.error(s"Page $page does not exist.")
       Sidebar.Page(Option.when(title.nonEmpty)(title), page, hidden)
     case RawInput(title, page, index, subsection, dir, hidden) if page.isEmpty && (!subsection.isEmpty() || !index.isEmpty()) =>
-      Sidebar.Category(Option.when(title.nonEmpty)(title), Option.when(index.nonEmpty)(index), subsection.asScala.map(toSidebar).toList, Option.when(dir.nonEmpty)(dir))
+      Sidebar.Category(Option.when(title.nonEmpty)(title), Option.when(index.nonEmpty)(index), subsection.asScala.map(toSidebar(_, content)).toList, Option.when(dir.nonEmpty)(dir))
     case RawInput(title, page, index, subsection, dir, hidden) =>
-      report.error(s"Error parsing YAML configuration file.\n$schemaMessage")
+      if title.isEmpty() then
```

Comment:
Why are we checking this? `title` is optional

Reply:
So it's optional for `subsection`, `- page` but not for a simple `page` from what I have seen in my tests. Example:
```
  - title: My title
    page: my-page1.md
```
But I noticed that it works with an index before (it's just ignored), so I'm thinking of adding `&& index.isEmpty`.
```
- index: my-page2.md
   page: my-page1.md
```

Reply:
Ho and this one works too:
```
index: index.md
subsection:
  - page: my-page1.md
    page: my-page2.md
  - page: my-page3.md
```
But the pages rendered are my-page2.md & my-page3.md